bn-apps/memberships-management/membership-service/src/test/kotlin/net/corda/businessnetworks/membership/BusinessNetworkAwareInitiatedFlowTest.kt

```diff
@@ -0,0 +1,132 @@
+
+        mockNetwork.runNetwork()
+
+        future.getOrThrow()
+    }
+
+    @Test
+    fun `active members should be able to transact on the business network`() {
+        val bnoNode = bnoNodes[0]
+
+        val participant1= participantsNodes[0]
+        val participant2= participantsNodes[1]
+
+        participant2.registerInitiatedFlow(BN_1_RespondingFlow::class.java)
+
+        runRequestAndActivateMembershipFlow(bnoNode, participant2)
+        runRequestAndActivateMembershipFlow(bnoNode, participant1)
+
+        val future = participant1.startFlow(BN_1_InitiatingFlow(participant2.identity()))
+
+        mockNetwork.runNetwork()
+
+        future.getOrThrow()
+    }
+
+    @Test(expected = NotAMemberException::class)
+    fun `non-members should be able to transact on the business network`() {
+        val bnoNode = bnoNodes[0]
+
+        val participant1= participantsNodes[0]
+        val participant2= participantsNodes[1]
+
+        participant2.registerInitiatedFlow(BN_1_RespondingFlow::class.java)
+
+        runRequestAndActivateMembershipFlow(bnoNode, participant2)
+
+        val future = participant1.startFlow(BN_1_InitiatingFlow(participant2.identity()))
+
+        mockNetwork.runNetwork()
+
+        future.getOrThrow()
+    }
+
+    @Test(expected = NotAMemberException::class)
+    fun `membership states with wrong membership contract should not be accepted`() {
+        val bnoNode = bnoNodes[0]
+
+        val participant1= participantsNodes[0]
+        val participant2= participantsNodes[1]
+
+        participant2.registerInitiatedFlow(BN_1_RespondingFlow::class.java)
+
+        runRequestAndActivateMembershipFlow(bnoNode, participant2)
+        runRequestAndActivateMembershipFlow(bnoNode, participant1)
+
+        // reloading configuration with a wrong contract name. All membership states verified by MembershipContract will be rejected
+        participant2.services.cordaService(MemberConfigurationService::class.java).reloadConfigurationFromFile(fileFromClasspath("membership-service-with-fake-contract-name.conf"))
+
+        val future = participant1.startFlow(BN_1_InitiatingFlow(participant2.identity()))
+
+        mockNetwork.runNetwork()
+
+        future.getOrThrow()
+    }
+}
+
+@InitiatingFlow
+class BN_1_InitiatingFlow(counterparty : Party) : AbstractDummyInitiatingFlow(counterparty)
+@InitiatingFlow
+class BN_2_InitiatingFlow(counterparty : Party) : AbstractDummyInitiatingFlow(counterparty)
+@InitiatingFlow
+class BN_3_InitiatingFlow(counterparty : Party) : AbstractDummyInitiatingFlow(counterparty)
+@InitiatingFlow
+class BN_4_InitiatingFlow(counterparty : Party) : AbstractDummyInitiatingFlow(counterparty)
+@InitiatingFlow
+class BN_5_InitiatingFlow(counterparty : Party) : AbstractDummyInitiatingFlow(counterparty)
+
+@InitiatedBy(BN_1_InitiatingFlow::class)
+class BN_1_RespondingFlow(session : FlowSession) : AbstractBNAwareRespondingFlow(session, "O=BNO_0,L=New York,C=US")
```

Comment:
As per discussed (and probably commented already, sorry!) although this is probably fine for now we will need a better way of linking the flow to a business network than passing the BNO's CX500 name.